### DISCUSSION TITLE
Getting Topics on a Specific DDS Domain

### DISCUSSION THREAD

**Discussion by mdavies2310** [OpenDDS/OpenDDS#4227] [Q&A] [answered]
Is there a way to get all topics on a specific dds domain? I only have the subscriber portion and am not getting data. Also is there a software manual that has all functions and descriptions?

**Comment by jrw972** [comment]
DDS has datareaders for "built-in topics" that describe discovered participants, subscriptions, publications, and topics.  See section 2.2.5 of the DDS Specification and https://opendds.readthedocs.io/en/latest-release/devguide/built_in_topics.html.

The DDS specification describes all of the API functions.  Additional OpenDDS-specific functions are described in the [DevGuide](https://opendds.readthedocs.io/en/latest-release/index.html).

**Reply by mdavies2310** [reply]
What would make the subscriber not getting to the on_data_available function? The domains are the same and the topics align(I think)

**Reply by mdavies2310** [reply]
In the DDS logs I can see places that reference the other computer that has the publisher.

**Reply by iguessthislldo** [reply, marked as answer]
> What would make the subscriber not getting to the on_data_available function? The domains are the same and the topics align(I think)

It could be a number of things. The discovery kind, transport kind, domain id, topic name, and topic type name all have to line up and the QoS needs to be compatible. That's also assuming the DataReader and Listener are set up correctly.

> In the DDS logs I can see places that reference the other computer that has the publisher.

If you post at least one of the logs we might be able to see why they're not associating.

**Reply by mdavies2310** [reply]
I will try to get one. We are using multicast and rtps_udp.

**Reply by mdavies2310** [reply]
I cannot get the file off but the main error I see is 
"ERROR: MulticastManager::process: ACE_SOCK_Dgram_Mcast::join_failed: unknown error -2"

**Reply by mdavies2310** [reply]
Ahh I figured it out.. I had modules in the IDL file and the publisher did not so they type names were off..
